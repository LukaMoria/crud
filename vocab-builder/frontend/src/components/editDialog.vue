<template>
    <q-dialog v-model="openEditDialog" persistent>
      <q-card style="min-width: 350px">
        <q-card-section>
          <div class="text-h6">Редактирование</div>
        </q-card-section>

        <q-card-section class="q-pt-none">
          <q-input dense v-model="localWord.english" autofocus @keyup.enter="prompt = false"></q-input>
        </q-card-section>

        <q-card-section class="q-pt-none">
          <q-input dense v-model="localWord.russian" autofocus @keyup.enter="prompt = false"></q-input>
        </q-card-section>

        <q-card-actions align="right" class="text-primary">
          <q-btn flat label="Отмена" @click="cancel"></q-btn>
          <q-btn flat label="Сохранить" @click="save"></q-btn>
        </q-card-actions>
      </q-card>
    </q-dialog>
</template>

<script>

export default {
  name:'EditDialog',
  props:{
    openEditDialog:{
      type: Boolean,
      required: true
    },
    currentEditableWord:{
      type:Object,
      required:true
    }
  },
  data(){
    return {
        localWord: Object.assign({},this.currentEditableWord)
    }
  },
  methods: {
    cancel () {
      this.$emit('close-popup', false)
    },
    save () {
      this.$emit('update-word', this.localWord)
    }
  },
  watch:{
    currentEditableWord: function(val){
      this.localWord = Object.assign({}, this.currentEditableWord)
    }
  }
}
</script>

<style>

</style>
