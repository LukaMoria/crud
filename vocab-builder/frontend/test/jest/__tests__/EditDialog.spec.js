
import { mount, createLocalVue, createWrapper, shallowMount } from '@vue/test-utils'
import editDialog from '../../../src/components/editDialog.vue'
import * as All from 'quasar'
// import langEn from 'quasar/lang/en-us' // change to any language you wish! => this breaks wallaby :(
const { Quasar } = All

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})

describe('editDialog.spec.js', () => {

    const localVue = createLocalVue()
    //Инициализация локального экземпляра Vue
    localVue.use(Quasar, { components }) 

    const createCmp = propsData => shallowMount(editDialog, { localVue, propsData });

    describe('Editable word params', () => {
        it('Checks identity editable word', () => {
            const cmp = createCmp({
                openEditDialog:true,
                currentEditableWord:{
                    russian: 'Дом',
                    english: 'House'
                }
            })
            expect(cmp.props('currentEditableWord').russian).toBe('Дом')
            expect(cmp.props('currentEditableWord').english).toBe('House')
        })

        it('Checks identity of prop and local Editable Word Russian', () => {
            const cmp = createCmp({
                openEditDialog:true,
                currentEditableWord:{
                    russian: 'Дом',
                    english: 'House'
                }
            })
            expect(cmp.vm.currentEditableWord.russian).toBe(cmp.props('currentEditableWord').russian)
        })

        it('Checks identity of prop and localWord English', () => {
            const cmp = createCmp({
                openEditDialog:true,
                currentEditableWord:{
                    russian: 'Дом',
                    english: 'House'
                }
            })
            expect(cmp.vm.currentEditableWord.english).toBe(cmp.props('currentEditableWord').english)
        })

        it('has a button save', () => {
            const cmp = createCmp({
                openEditDialog:true,
                currentEditableWord:{
                    russian: 'Дом',
                    english: 'House'
                }
            })
            expect(cmp.contains('#save')).toBe(true)
        })

        it('has a button cancel', () => {
            const cmp = createCmp({
                openEditDialog:true,
                currentEditableWord:{
                    russian: 'Дом',
                    english: 'House'
                }
            })
            expect(cmp.contains('#cancel')).toBe(true)
        })

        it('have a snapshot', () => {
            const div = document.createElement('div')
            document.body.appendChild(div)
            const cmp = mount(editDialog, {
                localVue,
                propsData:{
                    openEditDialog:true,
                    currentEditableWord:{
                        russian: 'Дом',
                        english: 'House'
                    }
                },
                attachTo: div,
            })

            expect(cmp.html()).toMatchSnapshot()
        })
    })
})