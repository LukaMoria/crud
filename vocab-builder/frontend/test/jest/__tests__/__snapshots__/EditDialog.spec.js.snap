// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editDialog.spec.js Checks shapshot 1`] = `<!---->`;

exports[`editDialog.spec.js Properties Checks shapshot 1`] = `<!---->`;
